iwyu.cc

```diff
@@ -2549,6 +2549,20 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
     return true;
   }
 
+  // The C++ spec does not allow us to apply typeid to an incomplete type.
+  // Therefore, we must report the type to be included.
+  bool VisitCXXTypeidExpr(clang::CXXTypeidExpr* expr) {
+    QualType type;
```

Comment:
Sorry I tricked you, we still need the `CanIgnoreCurrentASTNode` check here.

Reply:
```
    if (CanIgnoreCurrentASTNode())
      return true;
```

Reply:
Sorry about that - if I was paying attention, I should have spotted that myself :)

Reply:
OK, in the interest of nit-picky consistency, can you also lose the braces and move it first in the method?
```
bool VisitCXXTypeidExpr(clang::CXXTypeidExpr* expr) {
  if (CanIgnoreCurrentASTNode())
    return true;

  QualType type;
  ...
```

I'm too old to learn better tools (;-)), so I regularly use crazy pattern matching to find and rewrite code. Much easier if everything looks the same.

Reply:
Sure thing!